modin/engines/base/io/text/text_file_reader.py

```diff
@@ -56,218 +57,241 @@ def pathlib_or_pypath(cls, filepath_or_buffer):
-            max_bytes = float("inf")
+        tuple of bool and int,
+            bool: If file pointer reached the end of the file, but did not find
+            closing quote returns `False`. `True` in any other case.
+            int: Number of rows that was readed.
 
+        Notes
+        -----
+        The difference between `offset` is that `_read_rows` is more
+        specific version of `offset` which is focused of reading **rows**.
+        In common case it's better to use `offset`.
+        """
         if nrows is not None and nrows <= 0:
             return True, 0
 
-        # we need this condition to avoid unnecessary checks in `stop_condition`
-        # which executes in a huge for loop
-        if nrows is not None and max_bytes is None:
-            stop_condition = lambda rows_readed: rows_readed >= nrows  # noqa (E731)
-        elif nrows is not None and max_bytes is not None:
-            stop_condition = (
-                lambda rows_readed: f.tell() >= max_bytes or rows_readed >= nrows
-            )  # noqa (E731)
-        else:
-            stop_condition = lambda rows_readed: f.tell() >= max_bytes  # noqa (E731)
-
-        if max_bytes is not None:
-            max_bytes = max_bytes + f.tell()
-
         rows_readed = 0
-        outside_quotes = True
-        for line in f:
-            if is_quoting and line.count(quotechar) % 2:
-                outside_quotes = not outside_quotes
-            if outside_quotes:
-                rows_readed += 1
-                if stop_condition(rows_readed):
-                    break
 
+        if encoding not in ["unicode_escape", "utf16", "utf32"]:
+            for line in f:
+                if is_quoting and line.count(quotechar) % 2:
+                    outside_quotes = not outside_quotes
+                if outside_quotes:
+                    rows_readed += 1
+                    if rows_readed >= nrows:
+                        break
+        else:
+            buffer_size = io.DEFAULT_BUFFER_SIZE
+            chunk = f.read(buffer_size)
+            while chunk:
+                bytes_readed = 0
+                # split remove newline bytes from line
+                lines = chunk.split(newline)
+                chunk_size = len(chunk)
+                if len(lines) != 1:
+                    for line in lines[:-1]:
+                        bytes_readed += len(line) + len(newline)
+                        if is_quoting and line.count(quotechar) % 2:
+                            outside_quotes = not outside_quotes
+                        if outside_quotes:
+                            rows_readed += 1
+                            if rows_readed >= nrows:
+                                f.seek(f.tell() + bytes_readed - chunk_size)
+                                return outside_quotes, rows_readed
+
+                chunk = f.read(buffer_size)
+                if lines[-1]:
+                    # last line can be readed without newline bytes
```

Comment:
here and everywhere - past tense of "read" is also "read", as it's an irregular verb ;) pls fix

Reply:
fail :)